# HG changeset 9720 patch
From: kaf24@firebug.cl.cam.ac.uk
# Node ID ecae09fda29a9fc08059984b12259e1811a1444b
# Parent  ec8c43c9533249e4af781089b1d0423a20ad26c9
Subject: netfront: Turn grant-ref error into a fatal bug.
That's the best we can do until we have a backend driver recovery method.
Signed-off-by: Keir Fraser <keir@xensource.com>
xen-unstable changeset:   10275:3913a119477de17cf123991c113828437d7ba471
xen-unstable date:        Tue Jun 06 08:21:31 2006 +0100

Acked-by: Jan Beulich <jbeulich@novell.com>

Index: head-2006-06-07/drivers/xen/netfront/netfront.c
===================================================================
--- head-2006-06-07.orig/drivers/xen/netfront/netfront.c	2006-06-07 09:31:01.000000000 +0200
+++ head-2006-06-07/drivers/xen/netfront/netfront.c	2006-06-07 09:32:58.000000000 +0200
@@ -478,7 +478,7 @@ static void network_tx_buf_gc(struct net
 				printk(KERN_ALERT "network_tx_buf_gc: warning "
 				       "-- grant still in use by backend "
 				       "domain.\n");
-				goto out;
+				BUG();
 			}
 			gnttab_end_foreign_access_ref(
 				np->grant_tx_ref[id], GNTMAP_readonly);
@@ -504,7 +504,6 @@ static void network_tx_buf_gc(struct net
 		mb();
 	} while (prod != np->tx.sring->rsp_prod);
 
- out:
 	if (np->tx_full &&
 	    ((np->tx.sring->req_prod - prod) < NET_TX_RING_SIZE)) {
 		np->tx_full = 0;
